### DISCUSSION TITLE
Is there any tutorial to do unit test with goa?

### DISCUSSION THREAD

**Discussion by gmorinn** [goadesign/goa#3020] [Q&A] [answered]
I would like to know if it is possible to do unit tests using Goa design, either with Mocks or a test database.

**Comment by raphael** [comment, marked as answer]
There are examples in the [clue](https://github.com/goadesign/clue) repo, for example the [tests for the front service](https://github.com/goadesign/clue/blob/main/example/weather/services/front/front_test.go).

**Reply by gmorinn** [reply]
Oh thanks, I hadn't seen it. And good job with Goa Design!
